src/System.Management.Automation/CoreCLR/CorePsPlatform.cs

```diff
@@ -230,13 +230,14 @@ public static string SelectProductNameForDirectory(Platform.XDG_Type dirpath)
         {
             //TODO: XDG_DATA_DIRS implementation as per GitHub issue #1060
 
+            string defaultHomePath = System.Environment.GetEnvironmentVariable("HOME") ?? Directory.GetCurrentDirectory();
```

Comment:
For different permission between normal process and systemctl process, I am not sure it is by design or a OS bug, but we may only get this workaround currently.
If you have more idea, we can discuss.